presto-hive/src/test/java/com/facebook/presto/hive/TestCteExecution.java

```diff
@@ -65,10 +74,8 @@ public void testCteExecutionWhereOneCteRemovedBySimplifyEmptyInputRule()
                 " b AS ((SELECT orderkey FROM t) UNION (SELECT orderkey FROM t1)) " +
                 "SELECT * FROM b";
         QueryRunner queryRunner = getQueryRunner();
-        compareResults(queryRunner.execute(getMaterializedSession(),
-                        sql),
-                queryRunner.execute(getSession(),
-                        sql));
+        verifyResults(queryRunner, sql, ImmutableList.of(new CTEInfoValue("t", 1, false),
```

Comment:
doesn't matter a great deal but I tend to prefer Arrays.asList to ImmutableList.of to minimize thrid party dependencies

Reply:
"Prefer Immutable collections in Guava when possible." My experience is that it's not possible or advisable as much as most people think. I've had too many really ugly debugging sessions in production systems because an immutable type snuck into some code that expected to be able to mutate it. The root problem is that the Java Collections API was misdesigned 25 years ago in a way that violates the Liskov substitution principle, but we can't fix that here so the best we can do is not use types that don't implement the full List API. If we do want to use immutable types, we should make sure the API reflects that so folks aren't surprised when it doesn't mutate, but then we tightly couple to a third party API, which we also don't want to do. There's no good answer here.

All that being said, it's also possible to use Collections.unmodifiableList(Arrays.asList(...)) and retain immutability while avoiding third party dependencies. In this case since it's test code I'd prefer to keep it smaller as the immutability really isn't needed, but up to you.

Reply:
Sg, maybe we should update the guide

Reply:
+1 on "really ugly debugging sessions in production systems because an immutable type snuck into some code that expected to be able to mutate it", but as Jay mentioned it's explicitly called out in the Developer Guide that Immutable types should be used. Perhaps suggest an update to the Developer Guide for more nuisance use of the Immutable collections if we can come to a consensus.